wxWidgets/Qt occasional crash when file open dialog closes

Occasionally get this crash on linux - not consistently reproducible, looks like some issue with correctly initializing wxWidgets

Also strange that it seems to call QtWidgets.QFileDialog instead of the wxWidgets equivalent

Should check if it also happens sometimes on windows

```
pygame 2.1.2 (SDL 2.0.16, Python 3.9.7)
Hello from the pygame community. https://www.pygame.org/contribute.html
wx._core.wxAssertionError: C++ assertion ""Assert failure"" failed at /tmp/pip-install-9ngm6sal/wxpython_0f8c5d902fc246a9bb48e9905e41957d/ext/wxWidgets/src/gtk/dialog.cpp(196) in EndModal(): either wxDialog:EndModal called twice or ShowModal wasn't called

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/liam/.pyenv/versions/3.9.7/bin/motor_task_prototype", line 8, in <module>
    sys.exit(main())
  File "/home/liam/SSC/motor-task-prototype/src/motor_task_prototype/__main__.py", line 11, in main
    user_choices = mtp.setup()
  File "/home/liam/SSC/motor-task-prototype/src/motor_task_prototype/setup.py", line 122, in setup
    filename = fileOpenDlg(allowed="Psydat files (*.psydat)")
  File "/home/liam/.pyenv/versions/3.9.7/lib/python3.9/site-packages/psychopy/gui/qtgui.py", line 580, in fileOpenDlg
    filesToOpen = QtWidgets.QFileDialog.getOpenFileNames(parent=None,
SystemError: <built-in function getOpenFileNames> returned a result with an error set
1.5493  WARNING         We strongly recommend you activate the PTB sound engine in PsychoPy prefs as the preferred audio engine. Its timing is vastly superior. Your prefs are currently set to use ['sounddevice', 'PTB', 'pyo', 'pygame'] (in that order).
```

--- a/src/motor_task_prototype/setup.py
+++ b/src/motor_task_prototype/setup.py
@@ -3,7 +3,6 @@ from typing import List
 from typing import Optional
 
 import motor_task_prototype as mtp
-import wx
 from motor_task_prototype.display import get_display_options_from_user
 from motor_task_prototype.meta import get_metadata_from_user
 from motor_task_prototype.task import new_experiment_from_dicts
@@ -13,7 +12,9 @@ from motor_task_prototype.trial import get_trial_from_user
 from motor_task_prototype.trial import MotorTaskTrial
 from psychopy.data import TrialHandlerExt
 from psychopy.gui import fileOpenDlg
+from psychopy.gui.qtgui import ensureQtApp
 from psychopy.misc import fromFile
+from PyQt5 import QtWidgets
 
 
 @dataclass
@@ -31,42 +32,40 @@ def get_experiment_from_user() -> TrialHandlerExt:
         if len(trials) > 0:
             initial_trial = trials[-1]
         trials.append(get_trial_from_user(initial_trial))
-        dia_yes_no = wx.MessageDialog(
+        yes_no = QtWidgets.QMessageBox.question(
             None,
+            "Add another trial?",
             "Trials in experiment:\n\n"
             + describe_trials(trials)
             + "\n\nAdd another trial to the experiment?",
-            "Add another trial?",
-            style=wx.YES_NO,
+            QtWidgets.QMessageBox.Yes | QtWidgets.QMessageBox.No,
         )
-        result_yes_no = dia_yes_no.ShowModal()
-        dia_yes_no.Destroy()
-        if result_yes_no != wx.ID_YES:
+        if yes_no == QtWidgets.QMessageBox.Yes:
             add_another_trial = False
     display_options = get_display_options_from_user()
     return new_experiment_from_dicts(trials, display_options, metadata)
 
 
 def setup() -> Optional[UserChoices]:
-    app = wx.App(False)
-    app.MainLoop()
+    ensureQtApp()
     title = f"Motor Task Prototype {mtp.__version__}"
-    dialog = wx.SingleChoiceDialog(
+    options = [
+        "Create and run a new experiment",
+        "Import and run an existing experiment",
+        "Display existing experiment results",
+        "Exit",
+    ]
+    result, ok = QtWidgets.QInputDialog.getItem(
         None,
-        f"{title}\n\nhttps://ssciwr.github.io/motor-task-prototype/",
         title,
-        [
-            "Create and run a new experiment",
-            "Import and run an existing experiment",
-            "Display existing experiment results",
-            "Exit",
-        ],
+        f"{title}\n\nhttps://ssciwr.github.io/motor-task-prototype/",
+        options,
+        0,
+        False,
     )
-    result = dialog.ShowModal()
-    selection = dialog.GetSelection()
-    dialog.Destroy()
-    if result != wx.ID_OK:
+    if not ok:
         return None
+    selection = options.index(result)
     if selection == 0:
         return UserChoices(True, get_experiment_from_user())
     elif selection == 1:

--- a/src/motor_task_prototype/task.py
+++ b/src/motor_task_prototype/task.py
@@ -5,7 +5,9 @@ from typing import Union
 
 import numpy as np
 from motor_task_prototype import vis as mtpvis
+from motor_task_prototype.display import default_display_options
 from motor_task_prototype.geom import PointRotator
+from motor_task_prototype.meta import default_metadata
 from motor_task_prototype.meta import MotorTaskMetadata
 from motor_task_prototype.trial import MotorTaskTrial
 from motor_task_prototype.trial import validate_trial
@@ -25,6 +27,12 @@ from psychopy.visual.window import Window
 def new_experiment_from_trialhandler(experiment: TrialHandlerExt) -> TrialHandlerExt:
     for trial in experiment.trialList:
         validate_trial(trial)
+    if not experiment.extraInfo:
+        experiment.extraInfo = {}
+    if "display_options" not in experiment.extraInfo:
+        experiment.extraInfo["display_options"] = default_display_options()
+    if "metadata" not in experiment.extraInfo:
+        experiment.extraInfo["metadata"] = default_metadata()
     return TrialHandlerExt(
         experiment.trialList,
         nReps=1,

